src/chat.rs

```diff
@@ -1041,7 +1041,9 @@ impl Chat {
         if chat.id.is_archived_link() {
             chat.name = stock_str::archived_chats(context).await;
         } else {
-            if chat.typ == Chattype::Single {
+            if chat.typ == Chattype::Single && chat.name.is_empty() {
+                // chat.name is set to contact.display_name on changes,
+                // however, if things went wrong somehow, we do this here explicitly.
```

Comment:
this block was introduced at https://github.com/deltachat/deltachat-core-rust/pull/374 , not sure why, but as this is a highly visible area, it is probably better to have a second defense to avoid empty chat names :)